### DISCUSSION TITLE
how to integrate a new plugin into the solutions

### DISCUSSION THREAD

**Discussion by JiangGuotai** [mitsuba-renderer/mitsuba3#349] [Q&A] [answered]
I wrote a new plugin(sensor), but if i just simplily put it in the src/sensor and change the cmake file in the folder, the generated .sln cannot be complied correctly, so, are there some roadmaps if I wanna put my plugin in .sln and complie it correctly?

**Comment by njroussel** [comment, marked as answer]
This should work.

As you mentioned, only two steps are usually required:
1. Placing your new plugin in the `src/<plugin_type>` folder
2. Adding a `add_plugin(mynewplugin mynewplugin.cpp)` line to `src/<plugin_type>/CMakeLists.txt`

Could you share the error message you're getting? Ideally, also your new plugin source code.

**Reply by JiangGuotai** [reply]
Thank you!
I made some mistakes while compiling and now i fix it. 
Thanks for your time!
